crates/sui-bridge-indexer/src/eth_bridge_indexer.rs

```diff
@@ -33,20 +35,23 @@ use crate::{
 type RawEthData = (RawEthLog, Block<H256>, Transaction);
 
 pub struct EthSubscriptionDatasource {
-    bridge_address: EthAddress,
+    addresses: Vec<EthAddress>,
     eth_ws_url: String,
     indexer_metrics: BridgeIndexerMetrics,
 }
 
 impl EthSubscriptionDatasource {
     pub fn new(
-        eth_sui_bridge_contract_address: String,
+        eth_sui_bridge_contract_addresses: Vec<String>,
         eth_ws_url: String,
         indexer_metrics: BridgeIndexerMetrics,
     ) -> Result<Self, anyhow::Error> {
-        let bridge_address = EthAddress::from_str(&eth_sui_bridge_contract_address)?;
+        let bridge_addresses = eth_sui_bridge_contract_addresses
+            .iter()
+            .map(|address| EthAddress::from_str(address).unwrap())
+            .collect();
```

Comment:
unwrap here is unsafe. Do sth like this
```
        let bridge_addresses = eth_sui_bridge_contract_addresses
            .iter()
            .map(|address| EthAddress::from_str(address))
            .collect::<Result<_>>()?;
```
which returns the error as long as there is at least one error